lib/awful/hotkeys_popup/widget.lua

```diff
@@ -109,11 +109,60 @@ local widget = {
     group_rules = {},
 }
 
+-- To use special sortingm for hotkey modifiers, add the following line 
+-- your rc.lua file, after `local hotkeys_popup = require("awful.hotkeys_popup")`:
+--
+-- hotkeys_popup.widget.use_special_hotkey_mod_sort = true
+--
+-- And optionally, to customize the order, add, for example: 
+--
+-- hotkeys_popup.widget.special_hotkey_mod_order = {
+--         Shift = 1,
+--         Ctrl = 2,
+--         Super = 3,
+--         Alt = 4,
+-- }
```

Comment:
i think it would be better to have one parameter for this same thing (sorting), instead of two:
 - hotkeys_popup.widget.use_special_hotkey_mod_sort 
 - hotkeys_popup.widget.special_hotkey_mod_order